# Airbnb Barcelona — Análisis de Precios en Días de Semana

## Introducción

Este análisis explora el mercado de alojamientos de Airbnb en Barcelona 
durante días de semana. El objetivo es responder las siguientes preguntas:

- ¿Cómo varía el precio según el tipo de alojamiento?
- ¿Influye ser Superhost en el precio?
- ¿Existe relación entre la ubicación y el precio?
- ¿La limpieza y satisfacción del huésped impactan en el precio?

**Herramientas utilizadas:** Python · Pandas · Plotly  
**Dataset:** [Airbnb Prices in European Cities — Kaggle](https://www.kaggle.com/datasets/thedevastator/airbnb-prices-in-european-cities)

## 1. Importación de librerías

In [42]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import plotly.express as px # for visualization
import plotly.offline as py
import plotly.graph_objs as go
from plotly.figure_factory import create_table # for creating nice table

 1. Selección del dataframe

 Se selecciona el conjunto de dataframes de kaggle con el nombre "Airbnb Prices in European Cities" https://www.kaggle.com/datasets/thedevastator/airbnb-prices-in-european-cities) el cual evalua el precio de cada alojamiento
 de acuerdo a atributos como: Tipo de habitación, limpieza, puntuación, camas, distancia al centro y mas tanto durante los dias de semana como los fines de semana.

 Atributos:
*   realSum: (the total price of the listing) (Numeric)
*   room_type: (private, shared, entire home/apt) (Categorical)
*    room_shared:	Whether the room is shared or not. (Boolean)
*   room_private:	Whether the room is private or not. (Boolean)
*   person_capacity:	The maximum number of people that can stay in the room. (Numeric)
*   host_is_superhost: (boolean value indicating if host is a superhost or not)
*   multi: (indicator whether listing is for multiple rooms or not) (Boolean)
*   biz: (business indicator) Whether the listing is for business purposes or not. (Boolean)
*   cleanliness_rating:	The cleanliness rating of the listing. (Numeric)
*    guest_satisfaction_overall: (overall rating from guests camparing all listings offered by host
*   bedrooms: The number of bedrooms in the listing. (Numeric)
*   dist: (distance from city center) (Numeric)
*   metro_dist:	The distance from the nearest metro station. (Numeric)
*   lng:	The longitude of the listing. (Numeric)
*   lat:	The latitude of the listing. (Numeric)

Para este ejercicio se decide importar "barcelona_weekdays.csv".

## 2. Carga del dataset

Se utiliza el archivo `barcelona_weekdays.csv` descargado desde Kaggle.

In [43]:
# 1.1. Se importa el dataframe de Barcelona solo los dias de semana
import pandas as pd
bc_w = pd.read_csv("barcelona_weekdays.csv")
bc_w.head()

,Unnamed: 0,realSum,room_type,room_shared,room_private,person_capacity,host_is_superhost,multi,biz,cleanliness_rating,guest_satisfaction_overall,bedrooms,dist,metro_dist,attr_index,attr_index_norm,rest_index,rest_index_norm,lng,lat
0,0,474.317499,Entire home/apt,False,False,4.0,False,0,1,10.0,91.0,1,1.111996,0.630491,526.469420,17.942927,915.587083,20.154890,2.17556,41.39624
1,1,169.897829,Private room,False,True,2.0,True,1,0,10.0,88.0,1,1.751839,0.124017,320.127526,10.910462,794.277350,17.484489,2.14906,41.38714
2,2,161.984779,Private room,False,True,4.0,False,0,1,9.0,88.0,1,1.670493,0.080322,344.073936,11.726595,840.673617,18.505814,2.15357,41.37859
3,3,367.956804,Entire home/apt,False,False,3.0,False,0,1,10.0,91.0,1,1.475847,0.093107,400.057449,13.634603,946.589884,20.837357,2.16839,41.37390
4,4,196.895292,Private room,False,True,3.0,False,1,0,9.0,91.0,1,1.855452,0.272486,346.042245,11.793678,792.296039,17.440874,2.15238,41.37699


In [44]:
# 1.2. Se conocen los tipos de atributos que el dataframe contiene
bc_w.dtypes

,0
Unnamed: 0,int64
realSum,float64
room_type,object
room_shared,bool
room_private,bool
person_capacity,float64
host_is_superhost,bool
multi,int64
biz,int64
cleanliness_rating,float64


In [45]:
# 1.3. Se observan la cantidad de registros y atributos que el dataframe contiene
bc_w.shape

(1555, 20)

## 3. Limpieza de datos

In [46]:
#2. Se hace limpieza y orden al dataset
# 2.1. Se revisan los valores faltantes
bc_w.isna().sum()

,0
Unnamed: 0,0
realSum,0
room_type,0
room_shared,0
room_private,0
person_capacity,0
host_is_superhost,0
multi,0
biz,0
cleanliness_rating,0


In [47]:
# 2.2. Se revisan valores duplicados
bc_w.duplicated().sum()

np.int64(0)

In [48]:
# 2.3. Se realiza una descripción general de los datos de alojamientos en Barcelona los dias de semana agrupados por tipo de habitación.
(bc_w.groupby("room_type")[["realSum", "person_capacity", "cleanliness_rating","guest_satisfaction_overall", "bedrooms", "dist", "metro_dist"]].describe().T)


room_type                         Entire home/apt  Private room  Shared room
realSum                    count       362.000000   1185.000000     8.000000
                           mean        572.560054    202.754366   114.797403
                           std         556.749652     92.985757    28.260236
                           min         185.258454     69.588289    80.992390
                           25%         384.364745    150.580678    97.051225
                           50%         474.433868    185.258454   111.015430
                           75%         619.079759    229.245700   117.997533
                           max        6943.700980   1619.847790   171.294249
person_capacity            count       362.000000   1185.000000     8.000000
                           mean          4.480663      2.229536     2.750000
                           std           1.364737      0.614827     1.035098
                           min           2.000000      2.000000     2.000000
                           25%           4.000000      2.000000     2.000000
                           50%           4.000000      2.000000     2.000000
                           75%           6.000000      2.000000     4.000000
                           max           6.000000      6.000000     4.000000
cleanliness_rating         count       362.000000   1185.000000     8.000000
                           mean          9.196133      9.316456     8.750000
                           std           1.110357      0.973608     0.707107
                           min           2.000000      2.000000     8.000000
                           25%           9.000000      9.000000     8.000000
                           50%           9.000000     10.000000     9.000000
                           75%          10.000000     10.000000     9.000000
                           max          10.000000     10.000000    10.000000
guest_satisfaction_overall count       362.000000   1185.000000     8.000000
                           mean         88.720994     91.621097    88.375000
                           std          10.441079      7.994026     3.852179
                           min          40.000000     20.000000    83.000000
                           25%          84.250000     88.000000    86.250000
                           50%          91.000000     93.000000    88.500000
                           75%          96.000000     97.000000    90.250000
                           max         100.000000    100.000000    95.000000
bedrooms                   count       362.000000   1185.000000     8.000000
                           mean          1.842541      1.027004     1.000000
                           std           0.825536      0.248536     0.000000
                           min           0.000000      0.000000     1.000000
                           25%           1.000000      1.000000     1.000000
                           50%           2.000000      1.000000     1.000000
                           75%           2.000000      1.000000     1.000000
                           max           6.000000      3.000000     1.000000
dist                       count       362.000000   1185.000000     8.000000
                           mean          1.984475      2.147133     3.708951
                           std           1.014632      1.427652     1.926012
                           min           0.137703      0.119903     0.664513
                           25%           1.143773      1.059481     2.727817
                           50%           1.810257      1.720317     4.877805
                           75%           2.640847      3.056484     4.979719
                           max           5.807322      8.443957     5.134402
metro_dist                 count       362.000000   1185.000000     8.000000
                           mean          0.420574      0.437961     0.624993
                           std           0.274354      0.275135     0.390

#### Primeras Conclusiones

1.   Precio (realSum)
*   Los alojamientos completos son los más caros, con un precio medio muy elevado (572 €), casi 3 veces más que una habitación privada y 5 veces más que una compartida.
*   Las diferencias entre media y mediana en alojamientos completos (572 vs 474) indican una **asimetría positiva**, típica de mercados donde existen propiedades de lujo con valores muy altos.
*   El valor máximo (6943 €) confirma la presencia de **outliers extremos**, que elevan artificialmente el promedio.
*   **La dispersión** (std = 556 €) **es muy alta en Entire home/apt**, mientras que es mucho menor en habitaciones privadas, lo que sugiere un mercado de alojamientos completos más heterogéneo

2.   Capacidad y número de habitaciones
*   Los **alojamientos completos suelen tener más dormitorios y mayor capacidad**, lo que también explica parte del incremento del precio.
*   Las habitaciones privadas y compartidas mantienen estructuras más homogéneas

3.   Limpieza y satisfacción del huésped
*   **Las habitaciones privadas obtienen los mejores puntajes tanto en limpieza como en satisfacción**
*   Esto puede deberse a: espacios más pequeños, más fáciles de mantener limpios; interacción más directa del huésped con el anfitrión, expectativas más claras (un cuarto es más fácil de evaluar que una casa completa).
*   Las habitaciones compartidas presentan los valores más bajos, aunque el tamaño muestral de 8 limita la interpretación.

4.   Ubicación: distancia al centro y al metro
*   **Los alojamientos completos tienden a ubicarse más cerca del centro.**
*  ** Las habitaciones compartidas están más alejadas**, lo que coincide con estrategias orientadas a viajeros que priorizan precio por sobre ubicación.
*   **Tanto alojamientos completos como privados están relativamente bien conectados al metro** (< 0.45 km).
*   Esto sugiere dos segmentos de mercado: Turismo premium/parejas/familias, viviendas completas más céntricas; Turismo económico y jóvenes,  habitaciones privadas/compartidas más alejadas.













## 4. Visualizaciones

3.   Gráficos
*   Crea un gráfico de boxplot para diferentes categorías.
*   Incluye una breve explicación de las estadísticas mostradas en los boxplot de cada categoria (q1, mediana, etc..) y algunas conclusiones sobre las diferencias de los distintos grupos.




In [92]:
# 3.1. Boxplot Precio vs Tipo de Alojamiento
fig = px.box(
    bc_w,
    x="room_type",
    y="realSum",
    labels={
        "room_type": "Tipo de Alojamiento",
        "realSum": "Precio (€)"
    },
    title="Boxplot: Precio según Tipo de Alojamiento"
)

fig.show()


El boxplot revela que el precio se relaciona fuertemente con el tipo de alojamiento. Los alojamientos completos presentan un mercado mucho más heterogéneo y con presencia de propiedades de lujo que generan outliers extremos, mientras que las habitaciones privadas y compartidas mantienen un comportamiento más estable, simétrico y accesible.

In [ ]:
# 3.2. Boxplot Precio vs Superhost
fig = px.box(bc_w, x="host_is_superhost",
    y="realSum",
    labels={
        "host_is_superhost": "Es Superhost",
        "realSum": "Precio (€)"
    },
    title="Precio según si el anfitrión es Superhost"
)
fig.show()
fig.write_image("boxplot_precio_superhost.png")

*   No existe una diferencia marcada en el precio entre Superhosts y no-Superhosts:
A simple vista, las cajas (IQR), las medianas y los rangos de ambos grupos son bastante similares. Esto indica que ser Superhost no determina un precio significativamente mayor.
*   Ambos grupos presentan outliers, pero los no-superhosts tienen los más extremos. **Esto sugiere que los alojamientos premium, muy costosos o de lujo no necesariamente están gestionados por Superhosts.**
*   El precio típico (mediana) es similar, pero los extremos alteran el promedio en no-superhosts pero la presencia de outliers extremos en el grupo false podría elevar la media mucho más que la mediana.








In [ ]:
# 3.3. Boxplot Puntuación de Limpieza vs Tipo de alojamiento
fig = px.box(bc_w, x="room_type",
    y="cleanliness_rating",
    labels={
        "room_type": "Tipo de Alojamiento",
        "cleanliness_rating": "Puntuación de Limpieza (0 a 10)"
    },
    title="Limpieza según Tipo de Alojamiento"
)
fig.show()
fig.write_image("boxplot_limpieza_tipo.png")




*   Las habitaciones privadas tienen la puntuación de limpieza más alta y más
estable: La mediana es 10, el valor máximo posible, el rango intercuartílico (Q1–Q3) se mantiene entre 9 y 10, mostrando muy poca variabilidad; los outliers son pocos, moderados y se concentran entre 2 y 7, lo cual indica que la mayoría de estos alojamientos alcanzan un nivel de limpieza sobresaliente
*   Los alojamientos completos presentan mayor variabilidad en limpieza: La mediana es 9, similar a las privadas, pero el rango intercuartílico está entre 8 y 10. Esto puede deberse a:mayor tamaño del inmueble (baño, cocina, varias habitaciones), variabilidad en la gestión de limpieza según el anfitrión, alojamientos antiguos o menos mantenidos.
*   Las habitaciones compartidas tienen puntuaciones ligeramente inferiores: La mediana está entre 8 y 9 (visible en el rango de la caja), no se observan outliers extremos, pero esto también está influenciado por el muy bajo tamaño muestral (solo 8 registros).












In [ ]:
# 3.4. Distancia al centro de la ciudad vs Tipo de Alojamiento
fig = px.box(bc_w, x="room_type",
    y="dist",
    labels={
        "room_type": "Tipo de Alojamiento",
        "dist": "Distancia al centro de la ciudad (km)"
    },
    title="Distancia al Centro según Tipo de Alojamiento"
)
fig.show()
fig.write_image("boxplot_distancia_centro.png")




*   Los alojamientos completos (Entire home/apt) están más cerca del centro: La mediana está alrededor de 1.8–2 km, el IQR (Q1–Q3) es estrecho, mostrando valores bastante consistentes y sin valores extremos
*   Las habitaciones privadas muestran la mayor variabilidad en distancia: La mediana está alrededor de 1.7 km similar a los alojamientos completos, Sin embargo, la dispersión es mucho mayor. **Esto indica un segmento de habitaciones privadas ubicado tanto en zonas muy céntricas como muy alejadas.**
*   Las habitaciones compartidas son las más alejadas del centro: La mediana se ubica en torno a 4 km, desplazado hacia zonas más periféricas. No aparecen outliers hacia abajo lo que quiere decir que no hay habitaciones compartidas cerca del centro.





In [ ]:
# 3.5. Distancia al metro vs Tipo de Alojamiento
fig = px.box(bc_w, x="room_type",
    y="metro_dist",
    labels={
        "room_type": "Tipo de Alojamiento",
        "metro_dist": "Distancia a la estación de metro (km)"
    },
    title="Distancia al Metro según Tipo de Alojamiento"
)
fig.show()
fig.write_image("boxplot_distancia_metro.png")



*   **Los alojamientos completos y privados tienen accesibilidad similar** texto en negrita: La mediana está entre 0.3 y 0.4 km para ambas categorías, esto indica muy buena accesibilidad al transporte público en estas dos categorías.
*   Las habitaciones privadas tienen mayor dispersión:Hay numerosos outliers que llegan hasta los 2.4km, lo cual confirma nuevamente que el mercado de habitaciones privadas es muy variado en ubicación
*   **Las habitaciones compartidas están más alejadas del metro**: La mediana está claramente más alta, en torno a 0.6 km, El IQR también es mayor: entre 0.3 y 1 km. Los alojamientos compartidos tienen peor acceso al metro, de nuevo consistente con menor cercanía al centro y precios más bajos.







3. Gráfico de dispersión

*   Con tus datos, crea un gráfico de dispersión utilizando tanto el tamaño de los puntos como el color para representar dos dimensiones mas.
*    En el ejemplo de abajo serian el estado y la densidad de población. Comenta el gráfico y que conclusiones se pueden sacar de el




In [ ]:
#3.6. Scatter Plot Distancia al centro de la ciudad vs Precio. Color (Tipo de alojamiento), Tamaño de la burbuja (Puntuación General)
fig = px.scatter(bc_w,
    x="dist",
    y="realSum",
    log_y=True,
    color="room_type",
    size="guest_satisfaction_overall",
    hover_name="room_type",
    labels={
        "dist": "Distancia al centro de la ciudad (km)",
        "realSum": "Precio (€)",
        "room_type": "Tipo de Alojamiento",
        "guest_satisfaction_overall": "Puntuación General"
    },
    title="Scatter Plot Distancia al centro de la ciudad vs Precio"
)
fig.show()
fig.write_image("scatter_distancia_precio_satisfaccion.png")


*   Los precios más altos se concentran entre 1 y 3 km del centro. Los precios más altos se concentran entre 1 y 3 km del centro.
*   Los alojamientos completos (Entire home/apt) son los más caros y los más céntricos
*   Las habitaciones privadas son más baratas y están distribuidas en toda la ciudad
*   Las habitaciones compartidas son las más económicas y más alejadas
*   La puntuación general del huésped (tamaño de la burbuja) no muestra una relación fuerte con el precio ni con la distancia








El tamaño de las burbujas depende de la variable "guest_satisfaction_overall", las cuales son todas muy similares a 100%, motivo por el cual no se aprecian variaciones.

Se elige "cleanliness_rating" en su lugar que posee mas variacion:

In [ ]:
#3.7. Scatter Plot Distancia al centro de la ciudad vs Precio, color (Tipo de Alojamiento), Tamaño de la burbuja (Puntuación Limpieza).
fig = px.scatter(bc_w,
    x="dist",
    y="realSum",
    log_y=True,
    color="room_type",
    size="cleanliness_rating",
    hover_name="room_type",
    labels={
        "dist": "Distancia al centro de la ciudad (km)",
        "realSum": "Precio (€)",
        "room_type": "Tipo de Alojamiento",
        "cleanliness_rating": "Puntuación de limpieza (0 a 10)"
    },
    title="Scatter Plot Distancia al centro de la ciudad vs Precio (Limpieza)"
)
fig.show()
fig.write_image("scatter_distancia_precio_limpieza.png")


A diferencia del scatter anterior, aqui el tamaño de la burbuja se refiere a la Puntuacion de la Limpieza del alojamiento. **La puntuación de limpieza no muestra una relación fuerte con el precio ni con la distancia**
*   Se mantiene bastante similar entre alojamientos caros y económicos, también entre zonas céntricas y periféricas. Incluso alojamientos lejanos pueden tener limpieza 10/10, y alojamientos caros pueden tener puntuaciones más bajas.




In [74]:
bc_w["host_is_superhost"] = bc_w["host_is_superhost"].map({True: "Sí", False: "No"})

In [ ]:
#3.8. Scatter Plot Distancia al centro de la ciudad vs Precio, color (Superhost), Tamaño de la burbuja (Puntuación General).
fig = px.scatter(bc_w,
    x="dist",
    y="realSum",
    log_y=True,
    color="host_is_superhost",
    size="guest_satisfaction_overall",
    hover_name="room_type",
    labels={
        "dist": "Distancia al centro de la ciudad (km)",
        "realSum": "Precio (€)",
        "host_is_superhost": "Es superhost",
        "guest_satisfaction_overall": "Puntuación General (0 a 100)"
    },
    title="Scatter Plot Distancia vs Precio según Superhost"
)
fig.show()
fig.write_image("scatter_distancia_precio_superhost.png")

*   No se observa una diferencia significativa en el precio entre superhosts y no-superhosts
*   Tampoco hay una relación clara entre ser superhost y la distancia del centro
*   Superhosts y no-superhosts logran niveles de satisfacción similares
*   Superhosts aparecen en todos los tipos de propiedad.

Esto sugiere que ser superhost es más un indicador de buen historial y consistencia en la experiencia del huésped





In [69]:
bc_w["biz"] = bc_w["biz"].astype("category")

In [70]:
bc_w["biz"].dtype


CategoricalDtype(categories=[0, 1], ordered=False, categories_dtype=int64)

In [72]:
bc_w["biz"] = bc_w["biz"].map({1: "Sí", 0: "No"})

In [ ]:
#3.9. Scatter Plot Distancia al centro de la ciudad vs Precio, color (Apto para viaje de Negocios), Tamaño de la burbuja (Puntuación General).
fig = px.scatter(bc_w,
    x="dist",
    y="realSum",
    log_y=True,
    color="biz",
    size="guest_satisfaction_overall",
    hover_name="room_type",
    labels={
        "dist": "Distancia al centro de la ciudad (km)",
        "realSum": "Precio (€)",
        "biz": "Apto Viaje de Negocios",
        "guest_satisfaction_overall": "Puntuación General (0 a 100)"
    },
    title="Scatter Plot Distancia vs Precio según Viaje de Negocios"
)
fig.show()
fig.write_image("scatter_distancia_precio_negocios.png")

In [ ]:
#3.10. Scatter Plot Distancia al Puntuación General vs Precio, color (Tipo de Alojamiento), Tamaño de la burbuja (Puntuación Limpieza).
fig = px.scatter(bc_w,
    x="guest_satisfaction_overall",
    y="realSum",
    log_y=True,
    color="room_type",
    size="cleanliness_rating",
    hover_name="room_type",
    labels={
        "guest_satisfaction_overall": "Puntuación General (0 a 100)",
        "realSum": "Precio (€)",
        "room_type": "Tipo de Alojamiento",
        "cleanliness_rating": "Puntuación de limpieza (0 a 10)"
    },
    title="Scatter Plot Satisfacción General vs Precio"
)
fig.show()
fig.write_image("scatter_satisfaccion_precio.png")

*   No existe una diferencia significativa en precio entre alojamientos aptos para viaje de negocios y los que no lo son
*   Los alojamientos aptos para negocios no están más cerca del centro que los demás
*   El tamaño de las burbujas (satisfacción del huésped) es similar entre ambos grupos
*   La etiqueta “biz” puede ser más un atributo funcional que un indicador de calidad o precio





## 5. Conclusiones

- Los **alojamientos completos son los más caros y céntricos**, con alta variabilidad de precios
- Ser **Superhost no determina un precio mayor**, pero sí refleja consistencia en la experiencia
- Las **habitaciones compartidas son las más económicas y alejadas del centro**
- La **limpieza y satisfacción del huésped no muestran relación fuerte con el precio**
- Existen claramente **dos segmentos de mercado**: turismo premium (alojamientos completos céntricos) y turismo económico (habitaciones privadas/compartidas más alejadas)

---
*Análisis realizado por Juan Bautista Acuña — [GitHub](https://github.com/bautistaacuna)*